Modules/_datetimemodule.c

```diff
@@ -6537,136 +6514,131 @@ PyInit__datetime(void)
+        }                                           \
+        Py_DECREF(x);                               \
+    } while(0)
 
+    /* timedelta values */
+    PyObject *d = PyDateTime_DeltaType.tp_dict;
+    x = new_delta(0, 0, 1, 0);
+    DATETIME_ADD_MACRO(x, "resolution");
+    x = new_delta(-MAX_DELTA_DAYS, 0, 0, 0);
+    DATETIME_ADD_MACRO(x, "min");
     x = new_delta(MAX_DELTA_DAYS, 24*3600-1, 1000000-1, 0);
-    if (x == NULL || PyDict_SetItemString(d, "max", x) < 0)
-        return NULL;
-    Py_DECREF(x);
+    DATETIME_ADD_MACRO(x, "max");
 
     /* date values */
     d = PyDateTime_DateType.tp_dict;
-
     x = new_date(1, 1, 1);
-    if (x == NULL || PyDict_SetItemString(d, "min", x) < 0)
-        return NULL;
-    Py_DECREF(x);
-
+    DATETIME_ADD_MACRO(x, "min");
     x = new_date(MAXYEAR, 12, 31);
-    if (x == NULL || PyDict_SetItemString(d, "max", x) < 0)
-        return NULL;
-    Py_DECREF(x);
-
+    DATETIME_ADD_MACRO(x, "max");
     x = new_delta(1, 0, 0, 0);
-    if (x == NULL || PyDict_SetItemString(d, "resolution", x) < 0)
-        return NULL;
-    Py_DECREF(x);
+    DATETIME_ADD_MACRO(x, "resolution");
 
     /* time values */
     d = PyDateTime_TimeType.tp_dict;
-
     x = new_time(0, 0, 0, 0, Py_None, 0);
-    if (x == NULL || PyDict_SetItemString(d, "min", x) < 0)
-        return NULL;
-    Py_DECREF(x);
-
+    DATETIME_ADD_MACRO(x, "min");
     x = new_time(23, 59, 59, 999999, Py_None, 0);
-    if (x == NULL || PyDict_SetItemString(d, "max", x) < 0)
-        return NULL;
-    Py_DECREF(x);
-
+    DATETIME_ADD_MACRO(x, "max");
     x = new_delta(0, 0, 1, 0);
-    if (x == NULL || PyDict_SetItemString(d, "resolution", x) < 0)
-        return NULL;
-    Py_DECREF(x);
+    DATETIME_ADD_MACRO(x, "resolution");
 
     /* datetime values */
     d = PyDateTime_DateTimeType.tp_dict;
-
     x = new_datetime(1, 1, 1, 0, 0, 0, 0, Py_None, 0);
-    if (x == NULL || PyDict_SetItemString(d, "min", x) < 0)
-        return NULL;
-    Py_DECREF(x);
-
+    DATETIME_ADD_MACRO(x, "min");
     x = new_datetime(MAXYEAR, 12, 31, 23, 59, 59, 999999, Py_None, 0);
-    if (x == NULL || PyDict_SetItemString(d, "max", x) < 0)
-        return NULL;
-    Py_DECREF(x);
-
+    DATETIME_ADD_MACRO(x, "max");
     x = new_delta(0, 0, 1, 0);
-    if (x == NULL || PyDict_SetItemString(d, "resolution", x) < 0)
-        return NULL;
-    Py_DECREF(x);
+    DATETIME_ADD_MACRO(x, "resolution");
 
     /* timezone values */
     d = PyDateTime_TimeZoneType.tp_dict;
+    PyObject *delta = new_delta(0, 0, 0, 0);
+    if (delta == NULL) {
+        return -1;
+    }
 
-    delta = new_delta(0, 0, 0, 0);
-    if (delta == NULL)
-        return NULL;
     x = create_timezone(delta, NULL);
     Py_DECREF(delta);
-    if (x == NULL || PyDict_SetItemString(d, "utc", x) < 0)
-        return NULL;
+    if (x == NULL) {
+        return -1;
+    }
+    if (PyDict_SetItemString(d, "utc", x) < 0) {
+        Py_DECREF(x);
+        return -1;
+    }
```

Comment:
```suggestion
    if (x == NULL || PyDict_SetItemString(d, "utc", x) < 0) {
        Py_XDECREF(x);
        return -1;
    }
```

But DATETIME_ADD_MACRO() could be used here, no?

Reply:
@vstinner actually no, the macro does a DECREF at the end if it is successful.  Here we need to keep the ref count because we assign its value to CAPI.TimeZone_UTC

Reply:
Then I suggest to write:

```
x = ...;
if (x == NULL) {
    return -1;
}
PyDateTime_TimeZone_UTC = Py_NewRef(x);
DATETIME_ADD_MACRO(... x ...);
```